core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/operators/OperatorExecutor.scala

```diff
@@ -2,12 +2,15 @@ package edu.uci.ics.texera.workflow.common.operators
 
 import edu.uci.ics.amber.engine.common.tuple.amber.TupleLike
 import edu.uci.ics.amber.engine.common.workflow.PortIdentity
+import edu.uci.ics.texera.workflow.common.State
 import edu.uci.ics.texera.workflow.common.tuple.Tuple
 
 trait OperatorExecutor {
 
   def open(): Unit = {}
 
+  def processState(state: State, port: Int): State = state
```

Comment:
This seems to be default behavior of an executor to handle a state. when receiving a state, it does nothing (no change on its internal variable) but passes it downstream?

Instead, I think it should by default let state pass one operator at a time. So it should assign this state to its internal variables and consume the state (so do not pass it downstream). 

The other executors can overwrite this function as they need.

Reply:
I believe that an operator should, by default, pass its state downstream to support loop functionality. For example, the loop start could send a state i = 0 downstream, allowing any operator before the loop end to access the state i. If operators don't pass their state downstream by default, it would be impossible to propagate i automatically.

Reply:
PassToAllDownstream flag is introduced based on our discussion. By default, it's set to false, which means all states will only pass to one downstream operator, but users can still override this flag if they want.